lib/events/complete.go

```diff
@@ -169,6 +187,7 @@ func (u *UploadCompleter) checkUploads(ctx context.Context) error {
 		if err != nil {
 			if trace.IsNotFound(err) {
 				u.log.WithError(err).Warnf("Missing parts for upload %v. Moving on to next upload.", upload.ID)
+				incompleteSessionUploads.Dec()
```

Comment:
Why are we decreasing the counter in this case? Now this counter is no longer consistent with existing `completed` counter. Curious which one is more correct.

Reply:
This particular case (where you get a not found when trying to list parts) happens if you have multiple auth servers all racing eachother to complete the same set of uploads.

The `completed` counter is how many uploads _this_ instance of auth has completed, where the metric is attempting to provide the _total_ number of uncompleted uploads.

Technically we could remove this decrement. The metric would be slightly less accurate but it would correct itself next time the check runs.

The goal for this metric is not to know the exact number at any specific time - that would be pretty difficult in an HA setup. The goal is just to know if a large number of uncompleted uploads are accumulating and be able to see it go to 0 as they get completed.

Reply:
Will I remove this decrement then to keep it consistent with the complete count?

Reply:
Based on Zac's comment, it's more accurate doing the decrement here. So I think it's OK to leave it as it is. Maybe worth adding a comment.

Reply:
Agreed, let's keep it.